ordering/__init__.py

```diff
@@ -103,6 +104,14 @@ def assert_new_item(self, item: T) -> None:
         if item in self:
             raise KeyError("Ordering {} already contains {}".format(self, item))
 
+    @classmethod
+    def assert_unique_items(cls, items: Collection[_T]) -> None:
+        if len(items) > len(set(items)):
+            d: Dict[_T, bool] = {}
+            duplicates = [i for i in items if i in d or d.setdefault(i, False)]
+            raise ValueError('attempt to create {} containing duplicate items: {}'
```

Comment:
-  Personally I tend follow how the repr of the string looks, i.e. primarily using single quotes, then switching to double quotes when the string contains single quotes, finally switching back to single quotes if it contains both. However, I can see the logic in the convention you linked.<br><br>Also I tend to use triple single quotes for multiline strings that are not docstrings as depending on what editor I'm using it can allow for quickly differentiating between docstrings and multiline strings by their highlighting, even when the highlighter isn't directly aware of the concept of a docstring.

-  Okay, though I blame my non-capitalisation on the fact that unlike those in the standard library, most built-in error messages aren't capitalised.